cli/command/environment.go

```diff
@@ -217,42 +231,52 @@ func (e *EnvironmentCommand) update(c *cli.Context) error {
 }
 
 func (e *EnvironmentCommand) link(c *cli.Context) error {
-	args, err := extractArgs(c.Args(), "SOURCE_ENVIRONMENT_NAME", "DESTINATION_ENVIRONMENT_NAME")
-	if err != nil {
-		return err
-	}
+	generateAddLinkRequest := func(sourceEnvironmentID, destEnvironmentID string) (models.UpdateEnvironmentRequest, error) {
+		env, err := e.client.ReadEnvironment(sourceEnvironmentID)
+		if err != nil {
+			return models.UpdateEnvironmentRequest{}, err
+		}
 
-	sourceEnvironmentID, err := e.resolveSingleEntityIDHelper("environment", args["SOURCE_ENVIRONMENT_NAME"])
-	if err != nil {
-		return err
-	}
+		req := models.UpdateEnvironmentRequest{EnvironmentID: sourceEnvironmentID}
+		appendedLinks := append(env.Links, destEnvironmentID)
+		req.Links = &appendedLinks
 
-	destEnvironmentID, err := e.resolveSingleEntityIDHelper("environment", args["DESTINATION_ENVIRONMENT_NAME"])
-	if err != nil {
-		return err
+		return req, nil
 	}
 
-	if sourceEnvironmentID == destEnvironmentID {
-		return fmt.Errorf("Cannot link an environment to itself")
-	}
+	return e.updateEnvironmentLinksHelper(c, generateAddLinkRequest)
+}
 
-	req := models.CreateEnvironmentLinkRequest{
-		SourceEnvironmentID: sourceEnvironmentID,
-		DestEnvironmentID:   destEnvironmentID,
-	}
+func (e *EnvironmentCommand) unlink(c *cli.Context) error {
+	generateRemoveLinkRequest := func(sourceEnvironmentID, destEnvironmentID string) (models.UpdateEnvironmentRequest, error) {
+		env, err := e.client.ReadEnvironment(sourceEnvironmentID)
+		if err != nil {
+			return models.UpdateEnvironmentRequest{}, err
+		}
 
-	jobID, err := e.client.CreateLink(req)
-	if err != nil {
-		return err
+		destEnvIDIndex := -1
+		for i, link := range env.Links {
```

Comment:
see old example for how to delete while iterating over a list

Reply:
Oh yes! missed this :(. didn't really understand how your example was working so wanted to try it out in go playground before updating to it. pretty slick but still not sure how this works.
```
for i, link := range env.Links {
	if link == destEnvironmentID {
		links = append(env.Links[:i], env.Links[i+1:]...)
		i--
	}
}
```

https://play.golang.org/p/eWULErPdTD wouldn't deleting the last item always generate index out of bounds error?

Reply:
Oh I see now - looks like your example had an error. It should be 
```
links = append(env.Links[i:], env.Links[i+1:]...)

// instead of 
links = append(env.Links[:i], env.Links[i+1:]...)
```